### ISSUE TITLE
Binance: Fatal error on Api errors or data corruption in PHP

### ISSUE THREAD

**Issue by ashbeats** [ccxt/ccxt#1845]
- OS: **Windows 10 64bit,   Ubuntu 16.04.3 LTS**
- Programming Language version: **PHP 7.1.7**
- CCXT version: **1.10.1050**, **1.10.1063**
- Exchange: **Binance** (Not tested with any other exchanges)
- Method:  https://github.com/ccxt/ccxt/blob/73f2197a1dae7744ff3f096ef28d69576edd6cf2/php/binance.php#L308

`load_markets()` is used for this example to trigger it the chain of events. But the error is thrown in `Exchange::milliseconds() @ LN:309`.

-------------------------------------------------------------------

## Summary:

Adding property `options.timeDifference` to configs throws a Symfony\...\...\FatalThrowableError which **can't be caught using try/c.** 

```
$this->_exchange = new binance ([
    'enableRateLimit' => true,
    'timeout' => 30000,
    'options' => [
        "recvWindow" => 5000,
        "timeDifference" => 0,
        "adjustForTimeDifference" => false
    ],
]);
$mkts =  $this->_exchange->load_markets();
```
![issues_ccxt](https://user-images.githubusercontent.com/514003/36014164-0166146a-0d9b-11e8-8a36-18b37dbbae36.png)

**The error disappears** if I remove  `options.timeDifference`  from configs. 

```
$this->_exchange = new binance ([
    'enableRateLimit' => true,
    'timeout' => 30000,
    'options' => [
        "recvWindow" => 5000,
       /* "timeDifference" => 0, */
        "adjustForTimeDifference" => false
    ],
]);

```

## Request:

So my concern is not the fact that the error is thrown, but rather that a fatal error is thrown which can't be handled by the app. 

Most likely this error was caused by binance's api being down for a few hours. 

## Suggestion: 

CCXT should not throw fatal errors when the errors are triggered by data error's from the endpoint. Throw something catchable.

**Comment by ashbeats** [comment]
## Update:
Wrapping the constructor in `catch (\Throwable $e)` catches the general error with php 7.1.

**Comment by kroitor** [comment]
@ashbeats just fixed it in 1.10.1064. Thx for reporting!
